internal/query/projection/project_member.go

```diff
@@ -0,0 +1,131 @@
+package projection
+
+import (
+	"context"
+
+	"github.com/caos/logging"
+
+	"github.com/caos/zitadel/internal/errors"
+	"github.com/caos/zitadel/internal/eventstore"
+	"github.com/caos/zitadel/internal/eventstore/handler"
+	"github.com/caos/zitadel/internal/eventstore/handler/crdb"
+	"github.com/caos/zitadel/internal/repository/member"
+	"github.com/caos/zitadel/internal/repository/project"
+)
+
+type ProjectMemberProjection struct {
+	crdb.StatementHandler
+}
+
+const (
+	ProjectMemberProjectionTable = "zitadel.projections.project_members"
+)
+
+func NewProjectMemberProjection(ctx context.Context, config crdb.StatementHandlerConfig) *ProjectMemberProjection {
+	p := &ProjectMemberProjection{}
+	config.ProjectionName = ProjectMemberProjectionTable
+	config.Reducers = p.reducers()
+	p.StatementHandler = crdb.NewStatementHandler(ctx, config)
+	return p
+}
+
+func (p *ProjectMemberProjection) reducers() []handler.AggregateReducer {
+	return []handler.AggregateReducer{
+		{
+			Aggregate: project.AggregateType,
+			EventRedusers: []handler.EventReducer{
+				{
+					Event:  project.MemberAddedType,
+					Reduce: p.reduceAdded,
+				},
+				{
+					Event:  project.GrantMemberAddedType,
+					Reduce: p.reduceAdded,
+				},
+				{
+					Event:  project.MemberChangedType,
+					Reduce: p.reduceChanged,
+				},
+				{
+					Event:  project.MemberCascadeRemovedType,
+					Reduce: p.reduceCascadeRemoved,
+				},
+				{
+					Event:  project.MemberRemovedType,
+					Reduce: p.reduceRemoved,
+				},
+			},
+		},
+	}
+}
+
+type ProjectMemberColumn string
+
+const (
+	ProjectMemberProjectIDCol = "project_id"
+	ProjectMemberGrantIDCol   = "grant_id"
+)
+
+func (p *ProjectMemberProjection) reduceAdded(event eventstore.EventReader) (*handler.Statement, error) {
+	switch e := event.(type) {
+	case *project.MemberAddedEvent:
+		return reduceMemberAdded(e.MemberAddedEvent, withCol(ProjectMemberProjectIDCol, e.Aggregate().ID))
```

Comment:
doesnt this also trigger the project grant member? because there the aggregate id is the same

Reply:
yes it's filled into the same table. If it's a grant member the grantID is set

Reply:
you're right, I decided to split the projection into project_member and project_grant_member